File: src/paths.ts
import { homedir } from "node:os";
import { join } from "node:path";

const DATA_DIR_NAME = "clankers";
const DEFAULT_DB_FILE = "clankers.db";
const DEFAULT_CONFIG_FILE = "config.json";

export function getDataRoot(): string {
	if (process.env.CLANKERS_DATA_PATH) {
		return process.env.CLANKERS_DATA_PATH;
	}
	if (process.platform === "win32") {
		return process.env.APPDATA ?? join(homedir(), "AppData", "Roaming");
	}
	if (process.platform === "darwin") {
		return join(homedir(), "Library", "Application Support");
	}
	return process.env.XDG_DATA_HOME ?? join(homedir(), ".local", "share");
}

export function getDataDir(): string {
	return join(getDataRoot(), DATA_DIR_NAME);
}

export function getDbPath(): string {
	if (process.env.CLANKERS_DB_PATH) {
		return process.env.CLANKERS_DB_PATH;
	}
	return join(getDataDir(), DEFAULT_DB_FILE);
}

export function getConfigPath(): string {
	return join(getDataDir(), DEFAULT_CONFIG_FILE);
}

File: packages/cli/internal/paths/paths.go
package paths

import (
	"fmt"
	"os"
	"path/filepath"
	"runtime"
	"time"
)

const (
	dataDirName       = "clankers"
	defaultDBFile     = "clankers.db"
	defaultConfigFile = "clankers.json"
	defaultSocketName = "dxta-clankers.sock"
	logDirName        = "logs"
)

// Linux: $XDG_DATA_HOME or ~/.local/share
// macOS: ~/Library/Application Support
// Windows: %APPDATA% or ~/AppData/Roaming
// Can be overridden via CLANKERS_DATA_PATH.
func GetDataRoot() string {
	if v := os.Getenv("CLANKERS_DATA_PATH"); v != "" {
		return v
	}
	home, _ := os.UserHomeDir()
	switch runtime.GOOS {
	case "windows":
		if v := os.Getenv("APPDATA"); v != "" {
			return v
		}
		return filepath.Join(home, "AppData", "Roaming")
	case "darwin":
		return filepath.Join(home, "Library", "Application Support")
	default:
		if v := os.Getenv("XDG_DATA_HOME"); v != "" {
			return v
		}
		return filepath.Join(home, ".local", "share")
	}
}

func GetDataDir() string {
	return filepath.Join(GetDataRoot(), dataDirName)
}

func GetDbPath() string {
	if v := os.Getenv("CLANKERS_DB_PATH"); v != "" {
		return v
	}
	return filepath.Join(GetDataDir(), defaultDBFile)
}

func GetConfigPath() string {
	return filepath.Join(GetDataDir(), defaultConfigFile)
}

func GetSocketPath() string {
	if v := os.Getenv("CLANKERS_SOCKET_PATH"); v != "" {
		return v
	}
	if runtime.GOOS == "windows" {
		return `\\.\pipe\dxta-clankers`
	}
	return filepath.Join(GetDataDir(), defaultSocketName)
}

func GetLogDir() string {
	if v := os.Getenv("CLANKERS_LOG_PATH"); v != "" {
		return v
	}
	return filepath.Join(GetDataDir(), logDirName)
}

func GetCurrentLogFile() string {
	date := time.Now().Format("2006-01-02")
	return filepath.Join(GetLogDir(), fmt.Sprintf("clankers-%s.jsonl", date))
}
